modules/openapi-generator/src/main/resources/typescript-node/models.mustache

```diff
@@ -191,6 +191,13 @@ export class ApiKeyAuth implements Authentication {
             (<any>requestOptions.qs)[this.paramName] = this.apiKey;
         } else if (this.location == "header" && requestOptions && requestOptions.headers) {
             requestOptions.headers[this.paramName] = this.apiKey;
+        } else if (this.location == 'cookie' && requestOptions && requestOptions.headers) {
+            if (requestOptions.headers['Cookie']) {
+                requestOptions.headers['Cookie'] += '; ' + this.paramName + '=' + this.apiKey;
```

Comment:
Use encodeURIComponent() to encode the value

Reply:
Fixed and pushed update (rebased).